### ISSUE TITLE
[Bug]: Shellcheck ignore directives are placed on the wrong line in multiline pipes

### ISSUE THREAD

**Issue by Popkornium18** [chrisgrieser/nvim-rulebook#4]
### Bug Description

Assume we have this toy example:

```
dmesg |
  while read line; do
     echo "$line"
  done
```

Granted, writing this example that way looks stupid, but if you imagine `dmesg` being replaced with a longer command it makes sense to break up the pipe like that.

Shellcheck complains about the missing `-r` for the `read` command. Using rulebook to ignore this warning modifies the code like this:

```
dmesg |
  # shellcheck disable=SC2162
  while read line; do
     echo "$line"
  done
```

However it is not allowed to put comments into pipes like that. The correct solution would be:

```
# shellcheck disable=SC2162
dmesg |
  while read line; do
     echo "$line"
  done
```

### To Reproduce

See bug description

### Diagnostic Object

![image](https://github.com/chrisgrieser/nvim-rulebook/assets/46488511/c46c5e0a-e1f9-47bb-83e3-c94caea49a7a)

### neovim version

```Text
0.9.4
```

### Make sure you have done the following

- [X] I have updated to the latest version of the plugin.
